engine/src/ast/field_expr.rs

```diff
@@ -134,6 +134,20 @@ impl<'s> LhsFieldExpr<'s> {
             LhsFieldExpr::FunctionCallExpr(call) => call.uses(field),
         }
     }
+
+    pub fn compile_with<F: 's>(self, fnc: F) -> CompiledExpr<'s>
```

Comment:
I don't think it has to be public, given that it's just a private helper for the actual implementation.

Also nit: `func` would be slightly more readable :)

Reply:
Done :+1: